### PR TITLE
[Android][TASK][58882] Apply MVVM

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
- [#Task 58882: Apply MVVM](https://edu-redmine.sun-asterisk.vn/issues/58882)
## WHAT
- Apply MVVM 
## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://sal.vn/WtYPAy |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://sal.vn/piPoQn |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/vn/ztech/software/movie_streaming/utils/extensions/ViewExtension.kt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+fun ImageView.loadImage(
+    url: String,
+    @DrawableRes drawablePlaceHolder: Int = R.drawable.img_movie
+) {
+    Glide
+        .with(this)
+        .load(url)
+        .centerCrop()
+        .placeholder(drawablePlaceHolder)
```
Comment (hungld-1302) [inline app/src/main/java/vn/ztech/software/movie_streaming/utils/extensions/ViewExtension.kt:16]: thêm cả xử lý error nữa nha anh

File: app/src/main/java/vn/ztech/software/movie_streaming/utils/Constant.kt [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    const val NETWORK_READ_TIMEOUT = 30L
+    const val NETWORK_WRITE_TIMEOUT = 30L
+
+    enum class ScreenNumber {
+        Home,
+        Search,
+        Profile,
+    }
+
+}
```
Comment (hungld-1302) [inline app/src/main/java/vn/ztech/software/movie_streaming/utils/Constant.kt]: endline

File: app/src/main/java/vn/ztech/software/movie_streaming/base/BaseAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+import java.util.concurrent.Executors
+
+abstract class BaseAdapter<T, VB: ViewBinding, VH: BaseViewHolder<T, VB>>(diffCallBack: DiffUtil.ItemCallback<T>)
+    :ListAdapter<T, VH>(AsyncDifferConfig.Builder<T>(diffCallBack)
+        .setBackgroundThreadExecutor(Executors.newSingleThreadExecutor())
+        .build()
+    ){
+
+    override fun onBindViewHolder(holder: VH, position: Int) {
+        holder.bind(currentList[position])
```
Comment (hungvv-1301) [inline app/src/main/java/vn/ztech/software/movie_streaming/base/BaseAdapter.kt]:
```suggestion
        holder.bind(getItem(position))
```

File: app/src/main/java/vn/ztech/software/movie_streaming/data/model/Episode.kt [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package vn.ztech.software.movie_streaming.data.model
+
+data class Episode(
+    val id: String,
+    val number: Int,
+    val season: Int,
+    val title: String,
+    val url: String
```
Comment (ghost) [inline app/src/main/java/vn/ztech/software/movie_streaming/data/model/Episode.kt]:
```suggestion
data class Episode(
    val id: String?,
    val number: Int?,
    val season: Int?,
    val title: String?,
    val url: String?
```
anh check những file tương tự luôn nhé, cái này để tránh data bị null

**Review by hungld-1302 — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Review by hungvv-1301 — APPROVED** [review]

**Pull request merged**